File: 0110-balanced-binary-tree/0110-balanced-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    
   let check = (node) => {
       if(node == null) {
           return 0;
       } else {
           let lh = check(node.left);
           let rh = check(node.right);
           //console.log(node.val, lh, rh);
           if(Math.abs(lh - rh) <= 1) {
               return 1 + Math.max(lh, rh);
           } else {
               return Infinity;
           }
           
       }
   }
   
   return check(root) == Infinity ? false : true;
};

File: 0110-balanced-binary-tree/0110-balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int max = Integer.MAX_VALUE;
    public int check(TreeNode node) {
        if(node == null) {
            return 0;
        } else {
            int lh = check(node.left);
            int rh = check(node.right);
            if(lh == max || rh == max || Math.abs(lh - rh) > 1) {
                return max;
            } else {
                return 1  + Math.max(lh, rh);
            }
        }
    }
    
    public boolean isBalanced(TreeNode root) {
        return check(root) == max ? false : true;
    }
}